src/components/AddEditForm/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useContext } from "react";
+import "./AddEditForm.css";
+import { AddEditFormProps, Course } from "../../interfaces";
+import { DropdownFilterConstant } from "../../constants";
+import { MdClose } from "react-icons/md";
+import CoursesContext from "../../contexts/CoursesContext";
+
+const branchOptions = DropdownFilterConstant.branch.options
+  .filter((option) => {
+    return option !== "All Branches";
+  })
+  .map((option) => {
+    return <option>{option}</option>;
+  });
+
+const yearOptions = DropdownFilterConstant.year.options
+  .filter((option) => {
+    return option !== "All Years";
+  })
+  .map((option) => {
+    return <option value={option}>{option}</option>;
+  });
+
+function AddEditForm(props: AddEditFormProps) {
+  const isDisabled = props.type === "EDIT" ? true : false;
+
+  const [coursesState, dispatch] = useContext(CoursesContext);
+
+  const [filterState, setFilterState] = useState({
+    name: props.course.name,
+    code: props.course.code,
+    credits: props.course.credits,
+    branch: props.course.branch,
+    year: props.course.year,
+  });
+
+  const submitHandler = (event: any) => {
+    event.preventDefault();
+    const newCourse: Course = {
+      name: filterState.name,
+      prof: "John Doe",
+      strength: props.course.strength,
+      branch: filterState.branch,
+      year: filterState.year,
+      code: filterState.code,
+      credits: filterState.credits,
+    };
+
+    dispatch({ type: props.type, payload: newCourse });
+    props.closeFormHandler();
```

Comment:
Still not abstracted, additionally should be wrapped in `useCallback` and `event` type is missing.